### ISSUE TITLE
SC1010 false positive on ping -M do

### ISSUE THREAD

**Issue by TheCayman** [koalaman/shellcheck#3142]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1010
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:
```sh
# shellcheck=bash
ping -M do localhost
```

#### Here's what shellcheck currently says:
```
Line 2	SC1010: Use semicolon or linefeed before 'do' (or quote to make it literal).
```
#### Here's what I wanted or expected to see:
No error at all, as the 'do' here is a valid hint for ping and should not be interpreted by ShellCheck.

TODO: Describe expected/desired output
This is similar to issues https://github.com/koalaman/shellcheck/issues/1801 and https://github.com/koalaman/shellcheck/issues/1809.
1801 was not fixed, as that was related to a non-standard utility.
1809 was fixed.

This one is using ping, which is IMO a standard utility. The -M parameter specifies how ping should handle MTU path discovery. It can be passed three possible hints:
do
want
dont

See: https://linux.die.net/man/8/ping

The workaround of quoting the do-HINT does work, but without further explanation, it will not be clear to another developer why that was done, as it doesn't make sense in the context of the parameters to the ping command.
